helpers/util.sh

```diff
@@ -105,6 +105,45 @@ initlog() {
     [ -f "$LOG" ] && rm "$LOG"
 }
 
+get_last_tag() {
+    git describe --tags 2>/dev/null | sed -r "s/\-/\+/g"
+}
+
+get_package_version() {
+    pkg=$1
+    if [ "$(basename "$PWD")" != "$pkg" ]; then
+        die "get_package_version(): not within the $pkg directory"
+    fi
+    version=$(get_last_tag)
+    if [ -z "$version" ]; then
+        if [ "$BUILDOFFLINE" = "1" ]; then
+            die "Could not get version for $pkg: make sure it is cloned without --depth=1 or clone-depth=\"1\""
+        else
+            unshallow_attempt=$(git fetch --unshallow 2>&1)
+            ret=$?
+            if [ $ret -eq 128 ]; then
```

Comment:
Maybe better to check for non-0 return? so $ret -ne 0

Reply:
Done.